playbooks/upgrade.yml

```diff
@@ -0,0 +1,92 @@
+---
+- name: Upgrade testbed
+  hosts: all
+
+  vars:
+    ansible_galaxy: ~/venv/bin/ansible-galaxy
+    ansible_playbook: ~/venv/bin/ansible-playbook
+    basepath: "{{ ansible_user_dir }}/src/github.com/osism/testbed"
+    cloud_env: betacloud-ci
+    manager_address_file: "{{ basepath }}/terraform/.MANAGER_ADDRESS.{{ cloud_env }}"
+    manual_deploy: false
+    repo_path: "{{ ansible_user_dir }}/src/github.com"
+
+    version_ceph: pacific
+    version_ceph_next: pacific
+
+    version_manager: latest
+    version_manager_next: latest
+
+    version_openstack: xena
+    version_openstack_next: yoga
+
+  tasks:
+    - name: Make CI deployment with terraform
+      shell:
+        chdir: "{{ basepath }}/terraform"
+        cmd: make ENVIRONMENT={{ cloud_env }} VERSION_MANAGER={{ version_manager }} VERSION_CEPH={{ version_ceph }} VERSION_OPENSTACK={{ version_openstack }} create
+      when: not manual_deploy
+
+    - name: Fetch manager_host address
+      shell: cat "{{ manager_address_file }}"
+      register: manager_address
+
+    - name: Set manager_host address
+      set_fact:
+        manager_host: "{{ manager_address.stdout.split('=')[1] }}"
+
+    - name: Update ansible collections
+      shell:
+        chdir: "{{ ansible_user_dir }}"
+        cmd: |
+          {{ ansible_galaxy }} collection install --force "{{ ansible_user_dir }}/src/github.com/osism/ansible-collection-commons"
+          {{ ansible_galaxy }} collection install --force "{{ ansible_user_dir }}/src/github.com/osism/ansible-collection-services"
+      when: not manual_deploy
+
+    - name: Wait 300 seconds for port 22 to become open and contain "OpenSSH"
+      wait_for:
+        port: 22
+        host: '{{ manager_host }}'
+        search_regex: OpenSSH
+        delay: 10
+        timeout: 300
+
+    - name: Fetch manager ssh hostkey
+      shell: "sleep 60; ssh-keyscan {{ manager_host }} >> {{ ansible_user_dir }}/.ssh/known_hosts"
+
+    - name: Run part 0
+      shell:
+        chdir: "{{ basepath }}/ansible"
+        cmd: |
+          cp ../terraform/.id* .
+          {{ ansible_playbook }} -i ../terraform/inventory.{{ cloud_env }} --key-file .id_rsa.{{ cloud_env }} manager-part-0.yml -e cloud_env={{ cloud_env }} -e repo_path={{ repo_path }}
+
+    - name: Run part 1 + 2
+      shell:
+        chdir: "{{ basepath }}/ansible"
+        cmd: |
+          {{ ansible_playbook }} -i ../terraform/inventory.{{ cloud_env }} --key-file .id_rsa.{{ cloud_env }} manager-part-1.yml -e cloud_env={{ cloud_env }} -e repo_path={{ repo_path }}
+
+    - name: Run part 3
+      shell:
+        chdir: "{{ basepath }}/ansible"
+        cmd: "ssh -i .id_rsa.{{ cloud_env }} dragon@{{ manager_host }} /opt/configuration/scripts/deploy.sh"
+      when: not manual_deploy
+
+    - name: Run checks before the upgrade
+      shell:
+        chdir: "{{ basepath }}/terraform"
+        cmd: make ENVIRONMENT={{ cloud_env }} check
+      when: not manual_deploy
```

Comment:
Can you just import the deploy playbook for everything up to this? Would avoid a lot of duplication.